src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -517,6 +517,8 @@ private IEnumerable<LibraryDependency> GetLibraryDependenciesForCentralTransitiv
         {
             foreach (GraphNode<RemoteResolveResult> rootNode in targetGraph.Graphs)
             {
+                var dependencyDictionary = rootNode.Item.Data.Dependencies.ToDictionary(x => x.Name, x => x, StringComparer.OrdinalIgnoreCase);
```

Comment:
Are there cases where this dictionary won't end up being used?  On line 525 nodes can be skipped, so perhaps don't initialize this dictionary until its needed on line 538?

Reply:
Yeah, it is possible that the dictionary will not be eventually used but it is not very likely scenario. I don't have strong opinion here..